TimesSquare/TSQCalendarDayButton.m

```diff
@@ -8,31 +8,223 @@
+{
+    return self.subtitleLabel.font;
+}
+
+- (void)setSubtitleFont:(UIFont *)subtitleFont
+{
+    if (![self.subtitleFont isEqual:subtitleFont])
+    {
+        self.subtitleLabel.font = subtitleFont;
+        self.subtitleSymbolLabel.font = subtitleFont;
+        [self layoutSubviews];
+    }
+}
+
+- (UIColor *)subtitleColor
+{
+    return self.subtitleLabel.textColor;
+}
+
+- (void)setSubtitleColor:(UIColor *)subtitleColor
+{
+    self.subtitleLabel.textColor = subtitleColor;
+    self.subtitleSymbolLabel.textColor = subtitleColor;
+}
+
+- (UIImage *)icon
+{
+    return self.iconImageView.image;
+}
+
+- (void)setIcon:(UIImage *)icon
+{
+    if (![self.iconImageView.image isEqual:icon])
+    {
+        self.iconImageView.image = icon;
+        [self layoutSubviews];
+    }
+}
+
+- (CGSize)maxSubtitleSize
+{
+    CGFloat maxWidth = self.bounds.size.width - 2 * TSQCalendarRowCellSubtitleBuffer;
+    return CGSizeMake(maxWidth, TSQCalendarRowCellMaxSubtitleHeight);
+}
+
+- (CGSize)maxSubtitleSymboleSize
+{
+    return CGSizeMake(8.0f, TSQCalendarRowCellMaxSubtitleHeight);
+}
+
+- (void)layoutSubviews
+{
+    [super layoutSubviews];
+
+    CGFloat midX = CGRectGetMidX(self.bounds);
+    CGFloat subtitleCenterY = 0.75f * CGRectGetMaxY(self.bounds);
+
+    self.subtitleLabel.hidden = !([self.subtitleLabel.text length] > 0);
+    self.subtitleSymbolLabel.hidden = !([self.subtitleSymbolLabel.text length] > 0);
+    self.iconImageView.hidden = (self.iconImageView.image == nil);
+
+    CGFloat iconWidth = self.iconImageView.image.size.width;
+    CGFloat iconHeight = self.iconImageView.image.size.height;
+
+    if (! self.subtitleLabel.hidden)
+    {
+        CGSize maxSubtitleSize = [self maxSubtitleSize];
+        CGSize sizeThatFits = [self.subtitleLabel sizeThatFits:maxSubtitleSize];
+
+        CGFloat subtitleWidth = fminf(sizeThatFits.width, maxSubtitleSize.width);
+        CGFloat subtitleHeight = fminf(sizeThatFits.height, maxSubtitleSize.height);
+        CGFloat originX = midX - subtitleWidth/2.0f;
+        CGFloat originY = subtitleCenterY - subtitleHeight/2.0f;
+
+        self.subtitleLabel.frame = CGRectMake(originX,
+                                              originY,
+                                              subtitleWidth,
+                                              subtitleHeight);
+    }
+
+    if (! self.subtitleSymbolLabel.hidden)
+    {
+        CGSize maxSymbolSize = [self maxSubtitleSymboleSize];
+        CGSize sizeThatFits = [self.subtitleSymbolLabel sizeThatFits:maxSymbolSize];
+
+        CGFloat symbolWidth = fminf(sizeThatFits.width, maxSymbolSize.width);
+        CGFloat symbolHeight = fminf(sizeThatFits.height, maxSymbolSize.height);
+        CGFloat originX = CGRectGetMaxX(self.bounds) - TSQCalendarRowCellSubtitleBuffer;
+        CGFloat originY = subtitleCenterY - symbolHeight/2.0f;
+
+        if (! self.subtitleLabel.hidden)
+        {
+            // when subtitle is showing, shift symbol to right of subtitle
```

Comment:
Will the subtitleSymbolLabel ever be shown *without* the subtitleLabel?  Are you just technically supporting this in code, but not something we would use?